src/main/java/org/trebol/config/GlobalCorsConfiguration.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 The Trebol eCommerce Project
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+ * and associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the Software is furnished
+ * to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.trebol.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class GlobalCorsConfiguration {
+	
+	private final String[] allowedHeaders;
+	private final String[] allowedOrigins;
+	
+	@Autowired
+	public GlobalCorsConfiguration(CorsProperties corsProperties) {
+		this.allowedHeaders = corsProperties.getAllowedHeaders().split(corsProperties.getListDelimiter());
+	    this.allowedOrigins = corsProperties.getAllowedOrigins().split(corsProperties.getListDelimiter());		
+	}
+	
+	@Bean
+    public WebMvcConfigurer corsConfigurer() {
+        return new WebMvcConfigurer() {
+        	@Override
+        	public void addCorsMappings(CorsRegistry registry) {
+        		registry.addMapping("/public/login")
+        		   	.allowedOrigins(allowedOrigins)
+        		   	.allowedHeaders(allowedHeaders)
+        		   	.allowedMethods("POST");
+        		
+        		registry.addMapping("/public/guest")
+        			.allowedOrigins(allowedOrigins)
+        			.allowedHeaders(allowedHeaders)
+        			.allowedMethods("POST");
+        		
+        		registry.addMapping("/**")
```

Comment:
I'm not so confident about this mapping in particular... It basically means that any path is allowed to be queried.

Please, let's make sure it doesn't pose a security risk.

Reply:
So I added extra tests to see what will happen with other paths than what we are already using in the Controllers, basically testing for "any paths". I expected to see the tests to fail but actually they passed... (I tested them to be forbidden)

Then I realised that in the `GlobalCorsConfiguration`, the `.allowedMethods("")` actually clears the allowed methods for "any paths". So yes any paths are allowed but no methods are allowed? I wonder if this is safe? 😅 

```
registry.addMapping("/**")
        			.allowedOrigins(allowedOrigins)
        			.allowedHeaders(allowedHeaders)
        			.allowCredentials(true)
        			.maxAge(300L)
        			.allowedMethods("");
```

Reply:
Alright, I want to do some end-to-end tests as well.
Did you try separating multiple values with a semicolon `;` too?

Reply:
@bglamadrid Now yes 😅 still not working with multiple values... It supposed to work with comma `,`

I forgot to add to the above that I was trying to inject the properties to the `@CrossOrigin` annotation (in controllers) so this whole section could be removed from the `GlobalCorsConfiguration`:
```
registry.addMapping("/**")
        			.allowedOrigins(allowedOrigins)
        			.allowedHeaders(allowedHeaders)
        			.allowCredentials(true)
        			.maxAge(300L)
        			.allowedMethods("");
```

But unfortunately it didn't work for multiple values... If it would then the other properties like headers, credentials and max age could be injected too. [Spring docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/CrossOrigin.html)